packages/@aws-cdk/aws-ecs-patterns/lib/base/scheduled-task-base.ts

```diff
@@ -150,6 +162,7 @@ export abstract class ScheduledTaskBase extends CoreConstruct {
     }
     this.desiredTaskCount = props.desiredTaskCount || 1;
     this.subnetSelection = props.subnetSelection || { subnetType: SubnetType.PRIVATE };
+    this.securityGroups = props.securityGroups;
```

Comment:
What if the user does not provide this field? The doc comment says the a new security group will be created. At what point will this.securityGroups be assigned to the newly created default security group?

Reply:
Good question -- it's not totally obvious, but it gets created further down the stack within the events-target lib: https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-events-targets/lib/ecs-task.ts#L147-L150

Reply:
Okay thanks! I see now where the security group gets created. However, I did a quick test and after creating the security group, it is never assigned to the public securityGroups field of the ECS Task. Basically in the case where the customer omits the securityGroups prop:

```
const scheduledFargateTask = new ScheduledFargateTask(stack, 'ScheduledFargateTask', { 
  cluster, 
  scheduledFargateTaskImageOptions: {    
    image: ecs.ContainerImage.fromRegistry('amazon/amazon-ecs-sample'),    
    memoryLimitMiB: 512,
  },  
  schedule: events.Schedule.expression('rate(1 minute)'),  
});
```

then `scheduledFargateTask.securityGroups` will be undefined. Is that the intended behavior? It seems to me that this newly introduced securityGroups field should still give the user the ability to access the associated security group even if they didn't pass in a custom one.

Reply:
I noticed this PR will make the securityGroups available by surfacing the entire EcsTask: https://github.com/aws/aws-cdk/pull/14953/files. So I think we can remove this output property in ScheduledTaskBase.